pandas/tests/extension/test_arrow.py

```diff
@@ -248,8 +261,10 @@ def test_astype_str(self, data, request):
 class TestConstructors(base.BaseConstructorsTests):
     def test_from_dtype(self, data, request):
         pa_dtype = data.dtype.pyarrow_dtype
-        if (pa.types.is_timestamp(pa_dtype) and pa_dtype.tz) or pa.types.is_string(
-            pa_dtype
+        if (
+            (pa.types.is_timestamp(pa_dtype) and pa_dtype.tz)
+            or pa.types.is_string(pa_dtype)
+            or pa.types.is_decimal(pa_dtype)
```

Comment:
why doesn't this one work?

Reply:
`pa.type_for_alias` doesn't have an alias for `"decimal128..."` (similar to timestamps w/ tz) so this fails

Reply:
OK.  Can we get #50689 in and see if we can fix this case too